src/integer_mod_q/z_q/arithmetic/pow.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module provides an implementation of the [`Pow`] trait for [`Zq`].
+
+use crate::{
+    error::MathError,
+    integer::Z,
+    integer_mod_q::Zq,
+    macros::for_others::{implement_for_others, implement_for_owned},
+    traits::Pow,
+};
+use flint_sys::fmpz_mod::fmpz_mod_pow_fmpz;
+
+impl Pow<&Z> for &Zq {
+    type Output = Zq;
+
+    /// Raises the value of `self` to the power of an integer `exp`.
+    ///
+    /// Parameters:
+    /// - `exp`: specifies the exponent to which the value is raised
+    ///
+    /// Returns the value of `self` powered by `exp` as a new `Output` instance.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::integer::Z;
+    /// use qfall_math::integer_mod_q::Zq;
+    /// use qfall_math::traits::*;
+    ///
+    /// let base = Zq::try_from((2, 9)).unwrap();
+    /// let exp = Z::from(4);
+    ///
+    /// let powered_value = base.pow(&exp).unwrap();
+    ///
+    /// let cmp = Zq::try_from((7, 9)).unwrap();
+    /// assert_eq!(cmp, powered_value);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError::InvalidExponent`] if the provided exponent is
+    /// negative and the base value of `self` is not invertible.
+    fn pow(self, exp: &Z) -> Result<Self::Output, MathError> {
+        let mut out = self.clone();
+        if 0 == unsafe {
+            fmpz_mod_pow_fmpz(
+                &mut out.value.value,
+                &self.value.value,
+                &exp.value,
+                self.modulus.get_fmpz_mod_ctx_struct(),
+            )
+        } {
+            return Err(MathError::InvalidExponent(format!(
+                "The negative exponent {} was used for a non-invertible base {}",
+                exp, self
+            )));
+        }
+        Ok(out)
+    }
+}
```

Comment:
Please add the powers for the other types to the kanboard and mark those you are working on

Reply:
Already done

Reply:
I was taking about Z and Q, are they added? I have not found them

Reply:
Yes, it's called `Pow` in `To be discussed`